corehq/apps/api/resources/v0_5.py

```diff
@@ -1187,17 +1200,27 @@ def alter_list_data_to_serialize(self, request, data):
         data['meta']['local_date_timezone'] = self.api_params.local_timezone.zone
         data['meta']['total_count'] = self.count
 
-        params = request.GET.copy()  # Makes params mutable for creating next_url below
-
         if data['meta']['total_count'] > data['meta']['limit']:
+            original_params = request.GET.copy()  # Makes params mutable for creating next_url below
+            if 'cursor' in original_params:
+                params_string = b64decode(original_params['cursor']).decode('utf-8')
+                cursor_params = QueryDict(params_string, mutable=True)
+            else:
+                cursor_params = original_params.copy()
+                cursor_params.pop('limit', None)
+
             last_object = data['objects'][-1]
-            cursor = {
-                'cursor_local_date': last_object.data['local_date'],
-                'cursor_user': last_object.data['user'],
-            }
-            encoded_cursor = b64encode(urlencode(cursor).encode('utf-8'))
-            params['cursor'] = encoded_cursor
-            next_url = '{}?{}'.format(request.path, params.urlencode())
+            cursor_params['cursor_local_date'] = last_object.data['local_date']
+            cursor_params['cursor_user'] = last_object.data['user']
+            encoded_cursor = b64encode(urlencode(cursor_params).encode('utf-8'))
+
+            next_params = {'cursor': encoded_cursor}
+            # limit needs to directly be in request for tastypie to process it for pagination
```

Comment:
Partially, `limit` is being used to populate `data['meta']['limit']`. But alternatively, `data['meta']['limit']` can just be manually defined and I do think that makes it cleaner and easier to understand. [9d1fc69](https://github.com/dimagi/commcare-hq/pull/33238/commits/9d1fc69112a7e2541e95ab28b7b5465000566f78)

The only other consideration I had was that `limit` can safely be changed since it wouldn't affect pagination. But I guess there wouldn't be much of a reason to change it.